opentracing-kotlin/src/main/kotlin/org/zalando/opentracing/kotlin/TracerExtension.kt

```diff
@@ -0,0 +1,90 @@
+package org.zalando.opentracing.kotlin
+
+import io.opentracing.Scope
+import io.opentracing.Span
+import io.opentracing.SpanContext
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format
+import io.opentracing.propagation.TextMapAdapter
+import io.opentracing.tag.Tags
+
+inline fun <T : Tracer, R> T.trace(span: Span, block: (Span) -> R): R {
```

Comment:
```suggestion
inline fun <T : Tracer, R> T.activate(span: Span, block: (Span) -> R): R {
```

I'd remove the `span.finish()` call (see comment below) and then the whole purpose of this method would be scope activation.

One remark here: Even though this looks nice when it's being used, there are tons of cases where this method just simply won't be applicable.

I implemented tons of OpenTracing integrations and whenever you have a `Filter` or `Interceptor`-based approach, you need to split your code in half. There often is simply not a single block that you can put all your code in.

Reply:
Agreed, but currently these methods are simply for creating manual spans for important business logic which should not have anything like the interceptors/filters. We can create more functions that help with the use cases you mentioned.

Reply:
I agree with @piyushmor  - this is a library for application developers, not library authors.